tests/core/test_docstring_utils.py

```diff
@@ -0,0 +1,237 @@
+import pytest
+
+from ucai.core.utils.docstring_utils import DocstringInfo, parse_docstring
+
+
+def test_parse_docstring_empty():
+    """
+    Test that an empty docstring raises a ValueError.
+    """
+    with pytest.raises(ValueError, match="Docstring is empty"):
+        parse_docstring("")
+
+
+def test_parse_docstring_missing_description():
+    """
+    Test that a docstring without a description raises a ValueError.
+    """
+    docstring = """
+    Args:
+        x: The input value.
+    Returns:
+        int: The input value incremented by one.
+    """
+    with pytest.raises(ValueError, match="Function description is missing"):
+        parse_docstring(docstring)
+
+
+def test_parse_docstring_single_line_single_param():
+    """
+    Test parsing a simple docstring with a single-line description and one parameter.
+    """
+    docstring = """
+    Add one to the input.
+
+    Args:
+        x: The input value.
+    Returns:
+        int: The input value incremented by one.
+    """
+    expected = DocstringInfo(
+        description="Add one to the input.",
+        params={"x": "The input value."},
+        returns="int: The input value incremented by one.",
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_multiple_params():
+    """
+    Test parsing a docstring with multiple parameters.
+    """
+    docstring = """
+    Calculate the area of a rectangle.
+
+    Args:
+        width: The width of the rectangle.
+        height: The height of the rectangle.
+    Returns:
+        float: The area of the rectangle.
+    """
+    expected = DocstringInfo(
+        description="Calculate the area of a rectangle.",
+        params={"width": "The width of the rectangle.", "height": "The height of the rectangle."},
+        returns="float: The area of the rectangle.",
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_multi_line_param_description_with_colon():
+    """
+    Test parsing a docstring where a parameter description spans multiple lines and includes a colon.
+    """
+    docstring = """
+    Add one to the input.
+
+    Args:
+        x: The input value.
+            For example: 4
+    Returns:
+        int: The input value incremented by one.
+    """
+    expected = DocstringInfo(
+        description="Add one to the input.",
+        params={"x": "The input value. For example: 4"},
+        returns="int: The input value incremented by one.",
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_no_params():
+    """
+    Test parsing a docstring with no parameters.
+    """
+    docstring = """
+    Get the current timestamp.
+
```

Comment:
Can we also add test for
```
Args:
Returns:...
```

Reply:
yep, this correctly parses now (added tests too)